### ISSUE TITLE
Custom dataset from pandas dataframe throws IndexError

### ISSUE THREAD

**Issue by digital-thinking** [flairNLP/flair#1277]
I am trying to use flair for a custom pandas dataset, therefore I built a simple FlairDataset from pandas dataframes. My dataframe has a label and a text column, where I have a string and a binary label. I wonder how to get this simple thing running. I always got Indexing Errors, when trying to train a classifier on this.

```
from sklearn.model_selection import train_test_split
from flair.data import  Sentence, FlairDataset, segtok_tokenizer

class DataframeDataset(FlairDataset):
    def __init__(
        self,
        df,
        text_column,
        label_column,    
        max_tokens_per_doc=-1,
        max_chars_per_doc=-1,
        tokenizer=segtok_tokenizer,
    ):
        self.in_memory = True
        self.tokenizer = tokenizer
        self.sentences = []

        self.total_sentence_count: int = 0
        self.max_chars_per_doc = max_chars_per_doc
        self.max_tokens_per_doc = max_tokens_per_doc
        
        for _, row in df.iterrows():            
            line = row[text_column]            
            text = line.strip()
            
            labels = []
            labels.append(str(row[label_column]))
            
            if self.max_chars_per_doc > 0:
                text = text[: self.max_chars_per_doc]
            
            if text and labels:                
                sentence = Sentence(text, labels=labels, use_tokenizer=tokenizer)
                
                if len(sentence) > self.max_tokens_per_doc and self.max_tokens_per_doc > 0:
                    sentence.tokens = sentence.tokens[: self.max_tokens_per_doc]
                    
                if sentence is not None and len(sentence.tokens) > 0:                    
                    self.sentences.append(sentence)
                    self.total_sentence_count += 1
                
                        
        self.train, self.test = train_test_split(self.sentences, test_size=0.33, random_state=42)
        self.sentences = []
                        
dataset = DataframeDataset(df, 'message', 'status')

label_dictionary = Dictionary(add_unk=False)
label_dictionary.multi_label = False
label_dictionary.add_item('False')
label_dictionary.add_item('True')

word_embeddings = [#XLMEmbeddings('xlm-mlm-xnli15-1024'),
                   RoBERTaEmbeddings(),
                   # comment in flair embeddings for state-of-the-art results
                   # FlairEmbeddings('news-forward'),
                   # FlairEmbeddings('news-backward'),
                   ]

document_embeddings: DocumentRNNEmbeddings = DocumentRNNEmbeddings(word_embeddings,
                                                                     hidden_size=512,
                                                                     reproject_words=True,
                                                                     reproject_words_dimension=256,
                                                                     )

classifier = TextClassifier(document_embeddings, label_dictionary=label_dictionary)

trainer = ModelTrainer(classifier, dataset)

trainer.train('train_log',
              learning_rate=0.1,
              mini_batch_size=32,
              anneal_factor=0.5,
              patience=5,
              max_epochs=150,
              use_amp=False)
```
